### DISCUSSION TITLE
Reported Free Memory at startup

### DISCUSSION THREAD

**Discussion by Crazy-Charles** [mriscoc/Ender3V2S1#1191] [General]
Whilst loading the Official Build "Ender3V2-427-BLTUBL-MPC-20231217.bin" for some testing, the free memory is reported as :- 
"echo: Free Memory: 36703  PlannerBufferBytes: 1856"

Which means it has been built using the F103RC instead of F103RE as my very modified own firmware reports :- 
"echo: Free Memory: 54119  PlannerBufferBytes: 1728"

Quite a difference!

**Comment by mriscoc** [comment]
For universal compatibility, the firmware is compiled to work on both RCT and RET devices.

**Reply by Crazy-Charles** [reply]
Does the HAL lock the SRAM address space or will it just use all that it has?

**Reply by mriscoc** [reply]
The additional RAM space is unused because the buffer sizes were already defined. If you want to take advantage of additional RAM you need to increase the buffers size and compile for a RET6 SoC.
